File: candymarket.ui/src/Components/Candy/Candy.js
import React from 'react';

class Candy extends React.Component {
  getCandyId = () => {
    const candyId = this.props.candy.id;
    this.props.deleteCandy(candyId);
  }

  render() {
    const { candy } = this.props;
    return (
      <tr>
        <td>{candy.name}</td>
        <td>{candy.type}</td>
        <td>{candy.flavor}</td>
        <td>{candy.owner}</td>
        <td>
          <button className="btn btn-danger" onClick={this.getCandyId}>Delete</button>
        </td>
      </tr>
    );
  }
}

export default Candy;

File: CandyMarket.Api/DataModels/Candy.cs
using System;

namespace CandyMarket.Api.DataModels
{
    public class Candy
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Flavor { get; set; }
        public string Owner { get; set; }
        public bool IsDonate { get; set; }
    }
}
